applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -371,108 +371,33 @@ class GeometryUtilities
         KRATOS_TRY
 
         //resetting the normals
-        array_1d<double,3> zero = Vector(3);
-        noalias(zero) = ZeroVector(3);
-
-        for(auto & cond_i : rConditions)
-        {
-            Element::GeometryType& rNodes = cond_i.GetGeometry();
-            for(unsigned int in = 0; in<rNodes.size(); in++)
-                noalias((rNodes[in]).GetSolutionStepValue(NORMAL)) = zero;
-        }
-
-        //calculating the normals and storing on the conditions
-        array_1d<double,3> An;
-        if(dimension == 2)
+        const array_1d<double,3> zero = ZeroVector(3);
+        for(auto& cond_i : rConditions)
         {
-            for(auto & cond_i : rConditions)
+            for(auto& node_i : cond_i.GetGeometry())
             {
-                if (cond_i.GetGeometry().PointsNumber() == 2)
-                    CalculateNormal2D(cond_i,An);
+                noalias(node_i.GetSolutionStepValue(NORMAL)) = zero;
             }
         }
-        else if(dimension == 3)
+
+        //calculating the normals and summing up at nodes
+        const array_1d<double,3> local_coords = ZeroVector(3);
+        for(auto& cond_i : rConditions)
```

Comment:
same as above.

Reply:
Not sure how to implement the critical section for summing the nodal variable with he parallel_utils...

Reply:
Nevermind, i think i found an example